### PR TITLE
Enable customisation of client loop

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
@bilgili: override Client::clientLoop() with:
    while( isRunning( ))
    {
        if( idleFunc )
            idleFunc(); // order is important to latency
        processCommand( timeout );
    }

updateFunc is registered from Node::configInit:
   getClient()->setIdleFunction( std::bind( ... ))

Also call idleFunc() from application main loop.

File: eq/client.h [hunk 1]
```diff
@@ -102,6 +102,9 @@ class Client : public fabric::Client
      */
     EQ_API bool hasCommands();
 
+    /** @return true if the clientLoop() should keep running. */
```
Comment (tribal-tec) [inline eq/client.h]: missing @version

Comment (tribal-tec) [inline eq/client.h]: Shouldn't it be virtual? How does the customization work?

**Comment by tribal-tec** [comment]
Example/test should use this feature somewhere

File: eq/client.h (hunk 1 above)
Comment (eile) [inline eq/client.h]: No customisation is intended so far. The server will tell the client to stop.

**Comment by eile** [comment]
Example would need to launch at least one render client - do we really want that?

File: eq/client.h (hunk 1 above)
Comment (eile) [inline eq/client.h]: @version done

**Pull request merged**